### PR TITLE
Create cyclesort.c #2146

### LINKED ISSUE #2146: Cycle Sort
- [ ] C - @saloniankita
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @m-code12
- [x] Java
- [x] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by sejalshri**
implementation of cycle sort in C

Fixes #2146 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- implementation of cycle sort in C
- 

#### Languages Used: C
- 
- 

#### Files Added:
- 
[2146.txt](https://github.com/jainaman224/Algo_Ds_Notes/files/4537054/2146.txt)

- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by sejalshri** [comment]
kindly review this PR @jainaman224

**Comment by asha15** [comment]
please include the file in the relevant folder

**Comment by sejalshri** [comment]
done ma'am,@asha15 . please review it.

**Comment by sejalshri** [comment]
provided the proper indentation . @asha15 kindly check it.

File: Cycle_Sort/cyclesort.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#include <stdio.h>
+
+void main()
+{
+    int arr[100], i, j, k, p, n, value, index, l, temp;
+   	//input for array size
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:6]: remove the extra space

File: Cycle_Sort/cyclesort.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    printf("enter the number of elements required \n");
+    scanf("%d", &n);
+    printf("enter the elements of the array");
+    for (i = 0; i < n; i++)
+    {
+        scanf("%d", &arr[i]);
+    }
+   	//this sorting follows comparing process,with it's right elements hence loop goes to n-2 to avoid garbage values in the output
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:14]: please remove the extra space at the beginning

File: Cycle_Sort/cyclesort.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        index = k;
+       	//loop to find number of elements less than the element in consideration 
+        for (j = k + 1; j < n; j++)
+        {
+            if (arr[j] < value)
+            {
+                index++;
+            }
+        }
+       	//if the element is at right position
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:27]: correct the indentation

File: Cycle_Sort/cyclesort.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        if (index == k)
+        {
+            continue;
+        }
+       	//skipping duplicate elements
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:32]: correct the indentation

File: Cycle_Sort/cyclesort.c [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        while (arr[index] == value)
+        {
+
+            index++;
+        }
+       	//providing the correct position to the element
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:38]: correct the indentation

File: Cycle_Sort/cyclesort.c [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        }
+       	//if the element is at right position
+        if (index == k)
+        {
+            continue;
+        }
+       	//skipping duplicate elements
+        while (arr[index] == value)
+        {
+
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:35]: remove the extra line

File: Cycle_Sort/cyclesort.c [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            index++;
+        }
+       	//providing the correct position to the element
+        if (index != k)
+        {
+            temp = value;
+            value = arr[index];
+            arr[index] = temp;
+        }
+       	//rotate rest of the cycle
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:45]: correct the indentation

File: Cycle_Sort/cyclesort.c [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            {
+                if (arr[p] < value)
+                {
+                    index++;
+                }
+            }
+
+            while (arr[index] == value)
+            {
+
```
Comment (asha15) [inline Cycle_Sort/cyclesort.c:59]: remove the extra line

**Comment by sejalshri** [comment]
@asha15 , kindly review the changes.

**Comment by sejalshri** [comment]
done!

**Pull request merged**